### PR TITLE
metrics: fixes constant registering and unregistering of metrics map

### LINKED ISSUE #8550: Disable bpf map cache for cilium_metrics
If you run Cilium with debug logs, you'll find that there is constant registration/unregistration of the `cilium_metrics` map:

```
$ cilium_logs | grep "Registered.*cilium_metrics" | head -n 1
Jul 12 21:49:03 runtime1 cilium-agent[9159]: level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_metrics subsys=bpf
$ cilium_logs | grep "Registered.*cilium_metrics" | wc -l    
448
```
(This is over the period of about half an hour).

We should disable the map cache / registration,unregistration of the `cilium_metrics` map since the map is only populated from the BPF side so there's no use in using a cache with it.

### PR DESCRIPTION
**Pull request by fristonio**
Fixes #8550 

Previously during the periodic sync of metrics map with the Prometheus server, there was a constant registering and unregistering of cilium_metrics BPF map. This PR resolves this by using the once registered map for successive syncs.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8709)

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Review by brb — APPROVED** [review]
Nice!

**Comment by mrostecki** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

File: pkg/maps/metricsmap/metricsmap.go [hunk 1]
```diff
@@ -223,20 +223,29 @@ func updatePrometheusMetrics(key *Key, val *Value) {
...
+		// than opening the map again and again.
+		// This also prevents the constant registration and unregistration of the
+		// Map.
+		metricsMap, err = bpf.OpenMap(file)
+
+		if err != nil {
+			return fmt.Errorf("Unable to open metrics map: %s", err)
+		}
 	}
-	defer metricsmap.Close()
```
Comment (nirmoy) [inline pkg/maps/metricsmap/metricsmap.go:231]: LGTM with a minor concern: so the map never gets closed, I wonder if that will have any consequences. @brb WDYT ?

Comment (brb) [reply pkg/maps/metricsmap/metricsmap.go:231]: Good point, but AFAIK the map fd should get closed by the kernel when a cilium-agent process terminates. /cc @joestringer

**Pull request merged**